contrib/src/main/java/com/datatorrent/contrib/avro/PojoToAvro.java

```diff
@@ -0,0 +1,296 @@
+  int fieldErrorCnt = 0;
+
+  public final transient DefaultOutputPort<GenericRecord> output = new DefaultOutputPort<GenericRecord>();
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    try {
+      parseSchema();
+    } catch (IOException e) {
+      LOG.error("Exception in parsing schema");
+    }
+    initializeColumnMap(getSchema());
+  }
+
+  public void parseSchema() throws IOException
+  {
+    setSchema(new Schema.Parser().parse(getSchemaString()));
+  }
+
+  public String getSchemaString()
+  {
+    return schemaFile;
+  }
+
+  public void setSchemaString(String schemaFile)
+  {
+    this.schemaFile = schemaFile;
+  }
+
+  public Schema getSchema()
+  {
+    return schema;
+  }
+
+  public void setSchema(Schema schema)
+  {
+    this.schema = schema;
+  }
+
+  public List<Field> getColumnNames()
+  {
+    return columnNames;
+  }
+
+  public void setColumnNames(List<Field> columnNames)
+  {
+    this.columnNames = columnNames;
+  }
+
+  public Class<?> getCls()
+  {
+    return cls;
+  }
+
+  public void setCls(Class<?> cls)
+  {
+    this.cls = cls;
+  }
+
+  public List<Getter> getKeyMethodMap()
+  {
+    return keyMethodMap;
+  }
+
+  public void setKeyMethodMap(List<Getter> keyMethodMap)
+  {
+    this.keyMethodMap = keyMethodMap;
+  }
+
+  /**
+   * Adding this as a plug for being able to serialize non primitive types
+   * 
+   * @param -
+   *          object to serialize Returns a byte array
+   */
+
+  public byte[] serialize(Object obj) throws IOException
+  {
+    try (ByteArrayOutputStream b = new ByteArrayOutputStream()) {
+      try (ObjectOutputStream o = new ObjectOutputStream(b)) {
+        o.writeObject(obj);
+      }
+      return b.toByteArray();
+    }
+  }
+
+  /**
+   * @param -
+   *          className
+   * @param -
+   *          name of the field to create the getter for Returns a getter
+   */
+
+  private Getter<?, ?> generateGettersForField(Class<?> cls, String inputFieldName)
+      throws NoSuchFieldException, SecurityException
+  {
+    java.lang.reflect.Field f = cls.getDeclaredField(inputFieldName);
+    Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+
+    Getter<?, ?> classGetter = PojoUtils.createGetter(cls, inputFieldName, c);
+
+    return classGetter;
+  }
+
+  /**
+   * @param -
+   *          schema of the generic record Assumption is that the name of a
+   *          field in POJO is the same as the name in Avro schema
+   */
+
+  public void initializeColumnMap(Schema schema)
+  {
+    columnNames = schema.getFields();
+
+    keyMethodMap = new ArrayList<Getter>();
+    for (int i = 0; i < columnNames.size(); i++) {
+      try {
+        keyMethodMap.add(generateGettersForField(cls, columnNames.get(i).name()));
+      } catch (NoSuchFieldException | SecurityException e) {
+        throw new RuntimeException("Failed to initialize pojo class getters for field: ", e);
+      }
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = true, schemaRequired = true)
```

Comment:
Not sure if I understand the purpose of input port being optional.

If this is the only input port and optional to connect, why would anyone add this operator?